### PR TITLE
fixes failing test in Message

### PR DESCRIPTION
**Pull request by tantommy**
in `Message.fromCoseMessage`, `data` is set to the decoded cbor data per `content.set(4, cbor.decodeFirstSync(data));`
in `Message.fromObject`, `data` is cbor encoded per `content.set(4, cbor.encode(obj.data ? obj.data : new ArrayBuffer(0)));`

in the test we have:
```javascript
    const req = Message.fromObject(msg);
    const cbor = req.toCborData();
    expect(Message.fromCborData(cbor)).toStrictEqual(req);
```
the test is failing because the comparison between the `data` from `Message.fromCborData(cbor)` (calls `Message.fromCoseMessage`) and `Message.fromObject` is not the same because we are comparing the cbor encoded data in `req` to the cbor decoded data in `Message.fromCborData(cbor)`

### CODE REVIEW

**Comment by stanleyjones** [comment]
Since `Message.fromCborData` relies on `Message.fromCoseMessage` and we're already setting the `content` there, I think we can set the default data there, right? Then we might not have to change the tests.

**Comment by stanleyjones** [comment]
@tantommy Maybe we should chat about it... My understanding is that, in the test, `req` is going to be different from `fromCborMessage` because `fromCborMessage` will have an undefined `data` field. Is that right?

And I'm proposing that, in `Message` itself (not the test) we do something like this:

```ts
static fromCoseMessage(message: CoseMessage): Message {
...
const data = cbor.decodeFirstSync(data) : EMPTY_DATA;
content.set(4, data);
return new Message(content);
}
```

I admit I haven't tried it, but I think that would may `req` and `fromCborMessage` strictly equal? We could also do this in the constructor, but since we're already setting the data field here I thought it made the most sense.

**Comment by tantommy** [comment]
we have concluded that we will remove decoding of cbor from `Message.fromCoseMessage` per @stanleyjones and @hansl

**Review by hansl — APPROVED** [review]
LGTM, but maybe wait for @stanleyjones' review.

File: src/message/index.ts [hunk 1]
```diff
@@ -34,7 +34,9 @@ export class Message {
       content.set(2, obj.to.toString());
     }
     content.set(3, obj.method);
-    content.set(4, cbor.encode(obj.data ? obj.data : new ArrayBuffer(0)));
+    if (obj.data) {
+      content.set(4, cbor.encode(obj.data));
+    }
```
Comment (stanleyjones) [inline src/message/index.ts:39]: Don't we still need an empty buffer here if there's no `data` field in `obj`? That field is required by the spec, right @hansl ?

**Review by stanleyjones — APPROVED** [review]
One comment... might be blocking if the field is required, otherwise pre-approved.

**Comment by stanleyjones** [comment]
Never mind, it's optional: https://github.com/many-protocol/specification/blob/main/spec/cddl/many.cddl#L23

**Pull request merged**